src/api/app/controllers/status/concerns/set_checkable.rb

```diff
@@ -33,13 +29,26 @@ def set_repository
         return unless @project
 
         @checkable = @project.repositories.find_by(name: params[:repository_name])
+        @event_class = Event::StatusCheckForPublished
         return @checkable if @checkable
 
         @error_message = "Repository '#{params[:project_name]}/#{params[:repository_name]}' not found."
       end
 
+      def set_repository_architecture
```

Comment:
This concern really gets complicated :thinking: Can we maybe do something like 

``
set_project
set_repository
set_repository_architecture
set_bs_request
``

in set_checkable? I know this is not all from you but with this additional method it is just jumping from method to method....

Reply:
The problem is the error handling from what I see. What is an error for repository_architecture is the expected behaviour for set_repository. As such doing it from a concern is the problem IMO

Reply:
So what would you suggest instead a concern then?

Reply:
For now I simplified the code a little, check d0bf365